src/renderers/WebGLRenderer.js

```diff
@@ -1805,6 +1806,14 @@ class WebGLRenderer {
 
 					needsProgramChange = true;
 
+				} else if ( object.isBatchedMesh && materialProperties.batchingColor === true ) {
+
+					needsProgramChange = true;
+
+				} else if ( object.isBatchedMesh && materialProperties.batchingColor === false ) {
+
+					needsProgramChange = true;
+
```

Comment:
This condition doesn't look right - I might need to ask @Mugen87 for some insight into how this needs to work. Looking at the `InstanceMesh` color condition below it seems likely I'd misremembered how this portion of the code worked and my previous recommendation was incorrect. Sorry about that.

Reply:
Ok, bringing back the previous code.